Testexec search

jtluka opened the pull request.

jtluka committed:
PerfRepoTestExecution: add check for comment presence before getting value

Signed-off-by: Jan Tluka <jtluka@redhat.com>

jtluka committed:
add TestExecutionSearch object

Signed-off-by: Jan Tluka <jtluka@redhat.com>

olichtne commented on perfrepo-cli:
```diff
@@ -347,6 +354,55 @@ class TestExecCLI(GenericCLI):
+            elif argv[i] == "uid":
+                search_criteria.set_testUid(argv[i+1])
+                i += 2
+            elif argv[i] == "name":
```
Since this refers to a Test name (and the "uid" to Test uid) could we change the name of the parameter to something like tname/tuid? TestExecutions also have a 'name' attribute (though it's not a search criterion).

olichtne commented on perfrepo-cli:
```diff
@@ -347,6 +354,55 @@ class TestExecCLI(GenericCLI):
+            elif argv[i] == 'tags':
+                i += 1
+                while i < len(argv) and \
+                    argv[i] not in [ 'uid', 'name', 'param', 'id' ]:
```
The indentation of this condition could be better... I don't like when a multi-line condition starts
on the same indentation level as the inner code block

olichtne commented on perfrepo-cli:
```diff
@@ -347,6 +354,55 @@ class TestExecCLI(GenericCLI):
+                    argv[i] not in [ 'uid', 'name', 'param', 'id' ]:
+                    tag_str = argv[i]
+                    tags = tag_str.split(',')
+                    i += 1
```
This is a bit weird, the accepted syntax will look like:
...tags tag1,tag2  tag3 tag4,tag5

why not just parse tags the same way we do it for testexec create and accept just a single
comma separated tag string? The CLI help message also suggests this behaviour.

jtluka committed:
add search command for filtering testExecutions

Signed-off-by: Jan Tluka <jtluka@redhat.com>

jtluka commented:
Fri, Oct 21, 2016 at 10:11:21AM CEST, notifications@github.com wrote:

> olichtne commented on this pull request.
> 
> > -        i = 1
> >   +
> > -        search_criteria = perfrepo.PerfRepoTestExecutionSearch()
> >   +
> > -        # TODO: group, order-by, limit-from, how-many,
> > -        #       executed-before, executed-after
> >   +
> > -        while i < len(argv):
> > -            if argv[i] == "ids":
> > -                ids = argv[i+1].split(',')
> > -                search_criteria.set_ids(ids)
> > -                i += 2
> > -            elif argv[i] == "uid":
> > -                search_criteria.set_testUid(argv[i+1])
> > -                i += 2
> > -            elif argv[i] == "name":
> 
> Since this refers to a Test name (and the "uid" to Test uid) could we change the name of the parameter to something like tname/tuid? TestExecutions also have a 'name' attribute (though it's not a search criterion). 

Ok, will change it.

> > +
> > -        while i < len(argv):
> > -            if argv[i] == "ids":
> > -                ids = argv[i+1].split(',')
> > -                search_criteria.set_ids(ids)
> > -                i += 2
> > -            elif argv[i] == "uid":
> > -                search_criteria.set_testUid(argv[i+1])
> > -                i += 2
> > -            elif argv[i] == "name":
> > -                search_criteria.set_testName(argv[i+1])
> > -                i += 2
> > -            elif argv[i] == 'tags':
> > -                i += 1
> > -                while i < len(argv) and \
> > -                    argv[i] not in [ 'uid', 'name', 'param', 'id' ]:
> 
> The indentation of this condition could be better... I don't like when a multi-line condition starts
> on the same indentation level as the inner code block

Ok, I'll make things better.

> > -                ids = argv[i+1].split(',')
> > -                search_criteria.set_ids(ids)
> > -                i += 2
> > -            elif argv[i] == "uid":
> > -                search_criteria.set_testUid(argv[i+1])
> > -                i += 2
> > -            elif argv[i] == "name":
> > -                search_criteria.set_testName(argv[i+1])
> > -                i += 2
> > -            elif argv[i] == 'tags':
> > -                i += 1
> > -                while i < len(argv) and \
> > -                    argv[i] not in [ 'uid', 'name', 'param', 'id' ]:
> > -                    tag_str = argv[i]
> > -                    tags = tag_str.split(',')
> > -                    i += 1
> 
> This is a bit weird, the accepted syntax will look like:
> ...tags tag1,tag2  tag3 tag4,tag5

Err, not sure what kind of magic I wanted to do here. You're right, I'll
do it same as in texec.create.

Thanks for review.

> why not just parse tags the same way we do it for testexec create and accept just a single
> comma separated tag string? The CLI help message also suggests this behaviour. 
> 
> ## 
> 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/olichtne/python-perfrepo/pull/3#pullrequestreview-5214163
